src/plans/D3MAaveDaiPlan.sol

```diff
@@ -0,0 +1,213 @@
+        _;
+    }
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, uint256 data);
+    event File(bytes32 indexed what, address data);
+
+    constructor(address dai_, address pool_) public {
+        dai = TokenLike(dai_);
+        pool = LendingPoolLike(pool_);
+
+        // Fetch the reserve data from Aave
+        (,,,,,,, address adai_, address stableDebt_, address variableDebt_, address interestStrategy_,) = LendingPoolLike(pool_).getReserveData(dai_);
+        require(adai_ != address(0), "D3MAaveDaiPlan/invalid-adai");
+        require(stableDebt_ != address(0), "D3MAaveDaiPlan/invalid-stableDebt");
+        require(variableDebt_ != address(0), "D3MAaveDaiPlan/invalid-variableDebt");
+        require(interestStrategy_ != address(0), "D3MAaveDaiPlan/invalid-interestStrategy");
+
+        adai = adai_;
+        stableDebt = TokenLike(stableDebt_);
+        variableDebt = TokenLike(variableDebt_);
+        interestStrategy = InterestRateStrategyLike(interestStrategy_);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
+    }
+
+    // --- Math ---
+    uint256 constant RAY  = 10 ** 27;
+
+    function _add(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x + y) >= x, "D3MAaveDaiPlan/overflow");
+    }
+    function _sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, "D3MAaveDaiPlan/underflow");
+    }
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "D3MAaveDaiPlan/overflow");
+    }
+    function _rmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, y) / RAY;
+    }
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, RAY) / y;
+    }
+
+    // --- Admin ---
+    function file(bytes32 what, uint256 data) external auth {
+        if (what == "bar") {
+            require(data <= _maxBar(), "D3MAaveDaiPlan/above-max-interest");
+
+            bar = data;
+        } else revert("D3MAaveDaiPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        if (what == "interestStrategy") interestStrategy = InterestRateStrategyLike(data);
+        else revert("D3MAaveDaiPlan/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function _maxBar() internal view returns (uint256) {
+        return interestStrategy.getMaxVariableBorrowRate();
+    }
+
+    // --- Automated Rate targeting ---
+    function calculateTargetSupply(uint256 targetInterestRate) external view returns (uint256) {
+        uint256 totalDebt = _add(stableDebt.totalSupply(), variableDebt.totalSupply());
+        return _calculateTargetSupply(targetInterestRate, totalDebt);
+    }
```

Comment:
It looks like this is not needed anymore. @hexonaut can you confirm we don't need the external `calculateTargetSupply` anymore. I think it might have been an artifact of the helper

Reply:
Yeah if it's unused feel free to remove.

Reply:
done feb160d